From rain_maker@root-forum.org  Fri Sep 11 15:22:35 2009
From: "Axel K." <rain_maker@root-forum.org>
Date: Thu, 3 Sep 2009 20:53:36 +0200
Subject: Staging: rt2860: add new device ids
To: Bartlomiej Zolnierkiewicz <bzolnier@gmail.com>
Cc: Greg KH <gregkh@suse.de>
Message-ID: <200909032053.37543.rain_maker@root-forum.org>
Content-Disposition: inline


This patch adds new device IDs to ralink rt2860 driver in linux staging. The
device IDs were retrieved from the latest vendor release (version 2.1.2.0).

Signed-off-by: Axel Koellhofer <rain_maker@root-forum.org>
Acked-by: Bartlomiej Zolnierkiewicz <bzolnier@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/rt2860/2860_main_dev.c |    7 +++++++
 drivers/staging/rt2860/rt2860.h        |    2 ++
 2 files changed, 9 insertions(+)

--- a/drivers/staging/rt2860/2860_main_dev.c
+++ b/drivers/staging/rt2860/2860_main_dev.c
@@ -99,6 +99,13 @@ static struct pci_device_id rt2860_pci_t
 	{PCI_DEVICE(NIC_PCI_VENDOR_ID, NIC2760_PCI_DEVICE_ID)},
 	{PCI_DEVICE(NIC_PCI_VENDOR_ID, NIC2790_PCIe_DEVICE_ID)},
 	{PCI_DEVICE(VEN_AWT_PCI_VENDOR_ID, VEN_AWT_PCIe_DEVICE_ID)},
+	{PCI_DEVICE(EDIMAX_PCI_VENDOR_ID, 0x7708)},
+	{PCI_DEVICE(EDIMAX_PCI_VENDOR_ID, 0x7728)},
+	{PCI_DEVICE(EDIMAX_PCI_VENDOR_ID, 0x7758)},
+	{PCI_DEVICE(EDIMAX_PCI_VENDOR_ID, 0x7727)},
+	{PCI_DEVICE(EDIMAX_PCI_VENDOR_ID, 0x7738)},
+	{PCI_DEVICE(EDIMAX_PCI_VENDOR_ID, 0x7748)},
+	{PCI_DEVICE(EDIMAX_PCI_VENDOR_ID, 0x7768)},
     {0,}		// terminate list
 };
 
--- a/drivers/staging/rt2860/rt2860.h
+++ b/drivers/staging/rt2860/rt2860.h
@@ -292,6 +292,8 @@ rt2860_interrupt(int irq, void *dev_inst
 #define VEN_AWT_PCIe_DEVICE_ID	0x1059
 #define VEN_AWT_PCI_VENDOR_ID	0x1A3B
 
+#define EDIMAX_PCI_VENDOR_ID	0x1432
+
 // For RTMPPCIePowerLinkCtrlRestore () function
 #define RESTORE_HALT		    1
 #define RESTORE_WAKEUP		    2
